sdmetrics/visualization.py

```diff
@@ -317,13 +322,24 @@ def _generate_column_plot(
         fig = _generate_column_distplot(real_data, synthetic_data, plot_kwargs)
         trace_args = {'fill': 'tozeroy'}
 
-    for i, name in enumerate(col_names):
-        fig.update_traces(
-            x=pd.to_datetime(fig.data[i].x) if is_datetime_sdtype else fig.data[i].x,
-            hovertemplate=f'<b>{name}</b><br>Frequency: %{{y}}<extra></extra>',
-            selector={'name': name},
-            **trace_args,
-        )
+    annotations = []
+    if fig.data:
+        for i, name in enumerate(col_names):
+            fig.update_traces(
+                x=pd.to_datetime(fig.data[i].x) if is_datetime_sdtype else fig.data[i].x,
+                hovertemplate=f'<b>{name}</b><br>Frequency: %{{y}}<extra></extra>',
+                selector={'name': name},
+                **trace_args,
+            )
```

Comment:
can we rename the `i` to `idx` or `index` just to avoid having single characters